CHANGELOG.md

```diff
@@ -0,0 +1,30 @@
+# Gomobile-IPFS Global Changelog
+
+# Initial version
+
+## Golang Core
+
+- Provides basic bindings to go-ipfs
+- Provides a sockmanager that allows user to create UDS easily
+- Only supports simple types as input/output for request (byte[] or string)
+
+## Android/iOS Bridge
+
+- Provides a convenient IPFS Class that wraps underlying go objects:
+  - Repo path in configurable
+  - (Android) Repo storage is configurable: external or internal
+  - Basic methods that start, stop and restart the node
+  - NewRequest method that takes a command and returns a RequestBuilder
+- Provides a RequestBuilder Class:
+  - Simple bindings to the go-ipfs RequestBuilder
+  - Provides methods to set headers, options, arguments and body
+  - Method `send` returns a byte array
+  - (Android) Method `sendToJSONList` returns a JSON list
+  - (iOS) Method `sendToDict` returns a dict
+
+## Android/iOS Example Apps
+
+- Starts a node and display its peerID
+- Displays the number of connected peers
+- Provides a `Random XKCD` button that download a random XKCD from IPFS
+and displays it
```

Comment:
# Gomobile-IPFS Changelog

See https://github.com/ipfs-shipyard/gomobile-ipfs/releases

Reply:
hum, is this piece of code actually generating this file over time: https://github.com/ipfs-shipyard/gomobile-ipfs/pull/33/files#diff-56c30d9224897b08c28075052d4595c5R36-R39?

if yes, LGTM, else, keep things simple

Reply:
Yes the changelog is auto-generated by semantic-release/changelog, I simply created an initial version of the changelog to give users a minimum of information, then the CI will update it at each push on the master.

And the reason why I create a file instead of using only Github releases is for integration with Bintray which requires having a file on the repo.

You can check here (test from my fork) for a changelog updated by the CI and integrated into Bintray. https://bintray.com/beta/#/berty/gomobile-ipfs-android/core?tab=releaseNotes

Reply:
> A file containing only a link to the official change log is actually a file
:)

Honestly, now that the file is auto-generated with 4 lines of code in the CI config, I don't see the point of replacing the changelog with a link to a web page.

Especially if it allows to integrate cleanly in Bintray and potentially in other platforms in the future (yolo maybe one day for example, or any platform that would need a file).

In the same way, it is directly consultable in CLI by a dev that would have pulled the changes. Basically, from my point of view we don't gain anything in simplicity but it takes away possibilities.

Then if you really want to, I can delete it.